### PR TITLE
Allow multiple Editors

### PR DESCRIPTION
**Pull request by Knorrke**

### CODE REVIEW

File: packages/core/src/editor.tsx [hunk 1]
```diff
@@ -36,6 +53,54 @@ export function Editor<K extends string = string>(props: EditorProps<K>) {
...
+      {children}
+    </DragDropContextProvider>
+  )
+}
+
+export function EditorInstance<K extends string = string>(
+  props: EditorProps<K> & InnerEditorProps
+) {
+  const { store } = React.useContext(EditorContext)
+  const isMainInstance = React.useRef(
```
Comment (inyono) [inline packages/core/src/editor.tsx]: Never updated, so probably might not work as expected. Not sure what this is intended to do @Knorrke? I _suppose_ that you want the `React.useEffect` to depend on `props.mirror` instead and calculate `isMainInstance` inside the effect (might also depend on `store.getState()[props.scope]`?)

File: packages/core/src/editor.tsx [hunk 2]
```diff
@@ -36,6 +53,54 @@ export function Editor<K extends string = string>(props: EditorProps<K>) {
...
+    <DragDropContextProvider backend={HTML5Backend}>
+      {children}
+    </DragDropContextProvider>
+  )
+}
+
+export function EditorInstance<K extends string = string>(
+  props: EditorProps<K> & InnerEditorProps
+) {
+  const { store } = React.useContext(EditorContext)
```
Comment (inyono) [inline packages/core/src/editor.tsx]: Is this still correct?

File: packages/core/src/editor.tsx [hunk 3]
```diff
@@ -36,6 +53,54 @@ export function Editor<K extends string = string>(props: EditorProps<K>) {
...
+  const children = <Provider store={store}>{props.children}</Provider>
+  if (props.omitDragDropContext) return children
+  return (
+    <DragDropContextProvider backend={HTML5Backend}>
+      {children}
+    </DragDropContextProvider>
+  )
+}
+
+export function EditorInstance<K extends string = string>(
```
Comment (inyono) [inline packages/core/src/editor.tsx]: Review naming

File: packages/core/src/plugin-state/child.tsx [hunk 4]
```diff
@@ -17,10 +18,18 @@ function PluginPropsDocument<Props extends Record<string, unknown>>({
...
+  const { scope, editable } = React.useContext(ScopeContext)
   const pluginProps = React.useMemo(() => {
     return { ...props, parent: parentProps }
   }, [props, parentProps])
-  return <Document pluginProps={pluginProps} id={id} />
+  return (
+    <Document
+      pluginProps={pluginProps}
+      id={id}
+      scope={scope}
```
Comment (inyono) [inline packages/core/src/plugin-state/child.tsx]: Review usage of scope & editable. Might be easier to use the contexts in `Document` instead, not sure

File: packages/demo/src/container/index.tsx [hunk 5]
```diff
@@ -1,11 +1,12 @@
-import { Editor, EditorProps } from '@edtr-io/core'
```
Comment (inyono) [inline packages/demo/src/container/index.tsx]: Review that and use EditorProvider instead. Also note how the API feels in both cases (and improve it if needed)

File: scripts/changelog.ts [hunk 6]
```diff
@@ -259,15 +259,11 @@ async function exec(): Promise<void> {
         '**plugin-image**. Display Placeholder on empty image'
```
Comment (inyono) [inline scripts/changelog.ts]: Review changelog when done with rest

**Comment by inyono** [comment]
- [ ] Global bottom toolbar doesn't work correctly when using multiple editors (see: https://deploy-preview-136--romantic-kare-b2be52.netlify.com/?path=/story/article--multiple-editors).

File: packages/core/src/editor.tsx (hunk 1 above)
Comment (Knorrke) [reply packages/core/src/editor.tsx]: This was a hack... the problem is, that checking if the state was already initialized in `React.useEffect` (sometimes?) wouldn't work , as the state is initialized in another useEffect. So I wanted to check this during render time, but need it only once...

File: packages/core/src/editor.tsx (hunk 2 above)
Comment (Knorrke) [reply packages/core/src/editor.tsx]: yes, I think it is `{ store: { getState() }, dispatch() }` when using the EditorContext directly (not scoped)...

File: packages/core/src/editor.tsx (hunk 1 above)
Comment (inyono) [reply packages/core/src/editor.tsx]: Fixed in https://github.com/edtr-io/edtr-io/pull/136/commits/96fa20b65e5adb13e78d82da7d7b236373757aed

File: packages/core/src/plugin-state/child.tsx (hunk 4 above)
Comment (inyono) [reply packages/core/src/plugin-state/child.tsx]: Yep, that was easier (especially with the merged context containing both editable and scope), see https://github.com/edtr-io/edtr-io/pull/136/commits/cae9778852efa3f3b88ee1bd4b41a5643ddfb4b9

File: packages/core/src/editor.tsx (hunk 3 above)
Comment (inyono) [reply packages/core/src/editor.tsx]: **Single Document** (no changes)
```.tsx
<Editor {...props} />
```
**Multiple Documents** (EditorInstance → Document; Document → SubDocument (only used internally; Document is not editable by default)
```.tsx
<EditorProvider>
  <Document scope="main" editable />
  <Document scope="main" mirror />
</EditorProvider>
```

**Pull request merged**